src/app/content/components/Page/mediaModalManager.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { ReactNode, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import MediaModal from './MediaModal';
+import { HTMLElement, MouseEvent, KeyboardEvent, HTMLButtonElement, HTMLImageElement } from '@openstax/types/lib.dom';
+
+export function createMediaModalManager() {
+  let container: HTMLElement | null = null;
+  let setModalContent: ((content: ReactNode) => void) | null = null;
+
+  const open = (content: ReactNode) => {
+    setModalContent?.(content);
+  };
+
+// tslint:disable-next-line:variable-name
+  const MediaModalPortal = () => {
+    const [isOpen, setIsOpen] = React.useState(false);
+    const [modalContent, setContent] = React.useState<ReactNode>(null);
+
+    useEffect(() => {
+      setModalContent = (content) => {
+        setContent(content);
+        setIsOpen(true);
+      };
+      return () => { setModalContent = null; };
+    }, []);
+    if (typeof document === 'undefined') return null;
+
+    return isOpen ? createPortal(
+      <MediaModal isOpen={isOpen} onClose={() => setIsOpen(false)}>
+        {modalContent}
+      </MediaModal>,
+      document.body
+    ) : null;
+  };
+
+  const handleInteraction = (e: MouseEvent | KeyboardEvent) => {
+    const target = e.target as HTMLElement;
+    const button = target.closest('button.image-button-wrapper') as HTMLButtonElement | null;
+    if (!button) return;
+
+    if (e.type === 'keydown') {
+      const key = (e as KeyboardEvent).key;
+      if (key !== 'Enter' && key !== ' ') return;
+      e.preventDefault();
+    }
+
+    const img = button.querySelector('img') as HTMLImageElement | null;
+    if (!img) return;
+
+    open(
+      <img
+        tabIndex={0}
+        src={img.src}
+        alt={img.alt || ''}
+        width={img.width}
+        height={img.height}
+      />
+    );
+  };
+
+  const attachListeners = () => {
+    if (!container) return;
+    container.addEventListener('click', handleInteraction);
+    container.addEventListener('keydown', handleInteraction);
+  };
+
+  const detachListeners = () => {
+    if (!container) return;
+    container.removeEventListener('click', handleInteraction);
+    container.removeEventListener('keydown', handleInteraction);
+  };
+
+  const mount = (newContainer: HTMLElement) => {
+    // detach from previous if different container
+    if (container && container !== newContainer) {
+      detachListeners();
+    }
+
+    container = newContainer;
+    attachListeners();
```

Comment:
but if the container is the same container (which i think will happen when navigating between pages), you're attaching the listeners again each time mount is called right? aren't you getting duplicate listeners?

Reply:
The listeners won't be duplicated according to the [docs](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener). The browser dedupes listeners by type and listener reference. If the container is the same container which means it will have the same reference. It should not duplicate the listeners. So I believe this is safe. 
```
  const mount = (newContainer: HTMLElement) => {
    if (container !== newContainer) {
      detach();
      container = newContainer;
    }
    attach();
  };
```